File: Backend/src/main/java/com/example/demo/attendence/model/VacationRequestResponseModel.java
package com.example.demo.attendence.model;

import com.example.demo.attendence.entity.User;
import com.example.demo.attendence.utils.VacationStatus;
import com.example.demo.attendence.utils.VacationType;
import lombok.Data;

import java.time.LocalDate;

@Data
public class VacationRequestResponseModel {

    private Long id;
    private VacationType type;

    private VacationStatus status;

    private LocalDate startDate;

    private LocalDate endDate;

    private String comment;

    private int numberOfDays ;

    private UserResponseModel user;

}

File: UI/src/app/models/VacationRequestResponseModel.ts
import { VacationStatus } from "../utils/vacationStatus"
import { VacationType } from "../utils/vacationType"
import { UserModel } from "./UserModel";

export interface vacationRequestResponse{
    id:number;
    user:UserModel;
    type: VacationType

    status:VacationStatus

    startDate: Date

    endDate:Date

    comment:string

    numberOfDay:number
}